### PR TITLE
watchTermination should support async 'whenTerminated'

### PR DESCRIPTION
**Pull request by yim-lee**
Resolves https://github.com/apple/swift-distributed-actors/issues/930

### CODE REVIEW

File: Tests/DistributedActorsTests/LifecycleWatchTests.swift [hunk 1]
```diff
@@ -71,6 +71,22 @@ distributed actor Juliet: LifecycleWatch, CustomStringConvertible {
         }
     }
 
+    distributed func meetWatchAsyncCallback(
+        _ romeo: Romeo,
+        unwatch doUnwatch: Bool
+    ) async throws {
+        @Sendable
+        func asyncTerminated(_ terminatedIdentity: ClusterSystem.ActorID) async {
+            await self.probe.tell("Received terminated: \(terminatedIdentity)")
```
Comment (ktoso) [inline Tests/DistributedActorsTests/LifecycleWatchTests.swift:80]: This await seems not needed? Tell are unidirectional

**Review by ktoso — APPROVED** [review]
LGTM!

File: Tests/DistributedActorsTests/LifecycleWatchTests.swift (hunk 1 above)
Comment (yim-lee) [reply Tests/DistributedActorsTests/LifecycleWatchTests.swift:80]: Get compilation errors without `await`:

```
/code/Tests/DistributedActorsTests/LifecycleWatchTests.swift:80:13: error: expression is 'async' but is not marked with 'await'
            self.probe.tell("Received terminated: \(terminatedIdentity)")
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            await 
/code/Tests/DistributedActorsTests/LifecycleWatchTests.swift:80:13: note: property access is 'async'
            self.probe.tell("Received terminated: \(terminatedIdentity)")
```

Removing `@Sendable` gives warnings but compiles (without `await`). Should I keep `@Sendable`?

Comment (ktoso) [reply Tests/DistributedActorsTests/LifecycleWatchTests.swift:80]: Oh the probe access hmmm

Comment (ktoso) [reply Tests/DistributedActorsTests/LifecycleWatchTests.swift:80]: Checked out locally: the current code with await triggers a warning about `No 'async' operations occur within 'await' expression` removing the await "just worked" 🤔 

I think we're hitting some difference in toolchains? I'd like to try without the await and see what CI says, that'd be what I'd expect to work hmm

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
Huh I see, so it seems something changed in sendable semantics on latest nightlies... that's interesting -- merging your take on this API then @yim-lee !

**Pull request merged**